### PR TITLE
Koa 2: Upgrade to be a modern, promise-based (ctx, next) middleware

### PR DESCRIPTION
**Pull request by ide**
Updates this middleware to support Koa 2. The example server and readme file also reflect the modern API.

Test Plan:
 - `npm test`
 - `node example.js` and visited the page at localhost:3000 and saw that it sent 304s after soft-reloading the page in Chrome

### CODE REVIEW

**Comment by ide** [comment]
@jonathanong I wasn't able to submit a PR to a nonexistent `next` or `v2.x` branch but will resubmit this PR if you create one of those branches and want v2 development to happen there instead.

File: Readme.md [hunk 1]
```diff
@@ -28,14 +28,14 @@ app.use(etag());
...
-app.use(function *(next){
-  yield next;
-
-  this.body = {
-    name: 'tobi',
-    species: 'ferret',
-    age: 2
-  };
+app.use(function(ctx, next){
+  return next().then(function() {
```
Comment (coderhaoxin) [inline Readme.md]: It's better to use `async & await` for `examples`. FYI :)

**Comment by coderhaoxin** [comment]
land https://github.com/koajs/conditional-get/tree/next

**Comment by coderhaoxin** [comment]
@ide 

```
latest: 1.0.3
next: 2.0.0
```

**Comment by ide** [comment]
Thanks!

**Pull request closed without merging**